File: 171.cpp
// https://leetcode.com/problems/excel-sheet-column-number/

class Solution {
public:
    int titleToNumber(string s) {
        int sz = s.size();
        if (!sz) return 0;
        auto charToNum = [](char ch) { return ch - 'A' + 1; };
        return sz == 1 ? charToNum(s[0]) :
                        titleToNumber(s.substr(0, sz - 1)) * 26 + charToNum(s[sz - 1]);
    }
};

File: golang/171.go
package main

func titleToNumber(s string) int {
	l := len(s)
	if l == 0 {
		return 0
	}
	chartonum := func(b rune) int {
		return int(b - 'A' + 1)
	}
	ans := 0
	for _, v := range s {
		ans = ans*26 + chartonum(v)
	}
	return ans
}
